tools/sdmx_import/sdmx_client.py

```diff
@@ -37,36 +37,42 @@ def __init__(self, endpoint: str, agency_id: str):
         """
         self.agency_id = agency_id
         self.endpoint = endpoint
-        self.client = self._new_sdmx_client()
 
-    def _new_sdmx_client(self) -> sdmx.Client:
+    def _new_sdmx_client(self, agency_id: str) -> sdmx.Client:
         """
         Creates and configures an sdmx.Client for the specified endpoint and agency.
         """
-        source_id = self.agency_id
         custom_source = {
-            'id': source_id,
+            'id': agency_id,
             'url': self.endpoint,
-            'name': f'Custom source for {self.agency_id}'
+            'name': f'Custom source for {agency_id}'
         }
         sdmx.add_source(custom_source, override=True)
-        return sdmx.Client(source_id)
+        return sdmx.Client(agency_id)
 
-    def download_metadata(self, dataflow_id: str, output_path: str):
+    def download_metadata(self,
+                          dataflow_id: str,
+                          output_path: str,
+                          agency_id: str = None):
         """
         Fetches the complete metadata for a dataflow and saves it to a file as raw SDMX-ML (XML).
 
         Args:
             dataflow_id: The ID of the dataflow to retrieve
             output_path: Path where the metadata should be saved
+            agency_id: Optional. The ID of the agency providing the data. If
+                provided, this will override the agency_id set during client
+                initialization.
         """
         try:
             logging.info(
                 f"Fetching raw metadata for dataflow: {dataflow_id}...")
-            flow_msg = self.client.dataflow(dataflow_id,
-                                            agency_id=self.agency_id,
-                                            params={'references': 'all'},
-                                            tofile=output_path)
+            effective_agency_id = agency_id or self.agency_id
+            client = self._new_sdmx_client(effective_agency_id)
```

Comment:
Note: During my changes, I noticed that the overridden value is sometimes not being picked up and the old agency_id from client init was being used. So I decided to make this stateless and create a new client for every request.

I did a quick benchmark to measure the overhead of this change, and found it negligible (approximately 0.014 seconds per request).